### PR TITLE
create an example on how to run startup-scripts with a daemonset

### PR DESCRIPTION
**Pull request by mikedanese**
cc @aronchick @fabioy @roberthbailey

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/892)

### CODE REVIEW

File: startup-script/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM debian:jessie
```
Comment (roberthbailey) [inline startup-script/Dockerfile]: Is jessie just for convenience? It seems like we shouldn't need a fully fledged OS underneath a simple startup script example. (also, smaller base image == smaller attack surface => less image rebuilding as vulnerabilities as found)

File: startup-script/startup-script.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+spec:
+  template:
+    metadata:
+      labels:
+        app: startup-script
+    spec:
+      hostPID: true
+      containers:
+        - name: startup-script
+          image: gcr.io/mikedanese-k8s/startup-script:v1
```
Comment (roberthbailey) [inline startup-script/startup-script.yml]: From the above makefile it looks like the path here should be `gcr.io/google-containers/startup-script:v1`.

**Comment by roberthbailey** [comment]
Thanks for the PR mike; I didn't see it until the rh bot bounced it to the top of my inbox today.

**Comment by roberthbailey** [comment]
@mikedanese - do you want to finish this off and get it merged?

**Comment by test-foxish** [comment]
recomputing cla status...

File: startup-script/manage-startup-script.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+set -o nounset
+set -o pipefail
+
+CHECKPOINT_PATH="${CHECKPOINT_PATH:-/tmp/startup-script.kubernetes.io}"
+CHECK_INTERVAL_SECONDS="30"
+EXEC=(nsenter -t 1 -m -u -i -n -p --)
+
+do_startup_script() {
+  "${EXEC[@]}" bash -c "${STARTUP_SCRIPT}" && err=0 || err=$?
+  if [[ err != 0 ]]; then
```
Comment (jayme-github) [inline startup-script/manage-startup-script.sh]: Shouldn't this be `if [[ $err == 0 ]]; then`?

File: startup-script/manage-startup-script.sh [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    echo "!!! startup-script succeeded!"
+  else
+    echo "!!! startup-script failed!"
+  fi
+}
+
+while :; do
+
+  "${EXEC[@]}" stat "${CHECKPOINT_PATH}" && err=0 || err=$?
+  if [[ err != 0 ]]; then
```
Comment (jayme-github) [inline startup-script/manage-startup-script.sh]: And this `if [[ $err != 0 ]]; then`?

File: startup-script/Dockerfile (hunk 1 above)
Comment (jayme-github) [inline startup-script/Dockerfile]: Runs well for me using alpine ( with `bash` and `util-linux` package)

**Comment by tonglil** [comment]
This would be great to have!

**Comment by roberthbailey** [comment]
@mikedanese can we finish this up and merge it?

**Comment by tonglil** [comment]
I'd be happy to submit a copy if this one is unresponsive.

**Comment by mikedanese** [comment]
@roberthbailey @jayme-github @tonglil PTAL.

I pushed the image to gcr.io/google-containers.

**Comment by roberthbailey** [comment]
lgtm once the boilerplate is fixed.

**Comment by tonglil** [comment]
Does not seem to work anymore for 1.6? https://github.com/kubernetes/contrib/issues/2538

**Pull request merged**